server/src/main/java/org/elasticsearch/index/query/QueryShardContext.java

```diff
@@ -299,6 +299,18 @@ public SearchLookup lookup() {
         return this.lookup;
     }
 
+    private SearchLookup fetchLookup = null;
+
+    public SearchLookup fetchLookup() {
```

Comment:
I think this is the most controversial part of this change. Previously we were building the SourceLookup in the fetch phase and letting things that needed doc values at fetch time use the standard search lookup. This starts us down a road to removing that. I *think* we like this because we don't want things like the `explain` sub-phase to modify the rest of the fetch phases.

Reply:
Could we just create a `SearchLookup` within `FetchPhase`? I think we could store it on `HitContext` instead of storing `SourceLookup` directly. That would make its scope and lifetime clearer. It also feels weird to me to put this on `QueryShardContext`, since it's not meant to be used during the query phase.

Reply:
Hmmmmm. Let me dig into this again! I did it this way when I was building things really early and needed something in a much more "global" place to do the building.

I was trying not to modify `HitContext` so much - maybe it can be something we pass in to the phases when we build them. I'll take another look.

Reply:
We *can* create `SearchLookup` just for the `FetchPhase`. I'll push a change that does that. Sadly, we still have to delegate to `QueryShardContext` to create it because it is the thing with all of the right dependencies.

Reply:
I've pushed the change.

Reply:
I guess I need to read this discussion because I have been asking questions around this as part of my review :)